packages/query-core/src/infiniteQueryObserver.ts

```diff
@@ -38,9 +39,9 @@ export class InfiniteQueryObserver<
   TQueryKey
 > {
   // Type override
-  subscribe!: (
-    listener?: InfiniteQueryObserverListener<TData, TError>,
-  ) => () => void
+  subscribe!: Subscribable<
+    InfiniteQueryObserverListener<TData, TError>
+  >['subscribe']
```

Comment:
When doing type overriding, I want to use Subscribable and make it clearer to override while keeping the constraints of Subscribable, so I think it reduces the possibility of making constraint mistakes.

Reply:
shouldn't we do the same for other type overrides, like `fetch` a couple of lines below as well?

Reply:
I tried this on other type overrides(getCurrentResult, fetch) in [ccb4a62](https://github.com/TanStack/query/pull/7914/commits/ccb4a62a84e23bb8cc56ce6275033911de5978cc)